joblib/numpy_pickle.py

```diff
@@ -267,19 +267,14 @@ def save(self, obj):
                     # Pickling doesn't work with memmaped arrays
                     obj = self.np.asarray(obj)
                 return Pickler.save(self, obj)
-            self._npy_counter += 1
-            try:
+
+            if not obj.dtype.hasobject:
+                self._npy_counter += 1
                 filename = '%s_%02i.npy' % (self._filename,
                                             self._npy_counter)
                 # This converts the array in a container
                 obj, filename = self._write_array(obj, filename)
                 self._filenames.append(filename)
-            except:
```

Comment:
I don't remember.

However, I do know something: the policy of joblib is to try to never fail. Our philosophy is that adding joblib to your code should not making more fragile. That's why try / excepts are sprinkled in the codebase. The idea is that we might not think about all the possible errors, so if we can't persist the fast way, we persist the standard way.

I am afraid that joblib is becoming more brittle as it is becoming more sophisticated.